app/grandchallenge/algorithms/forms.py

```diff
@@ -205,11 +230,109 @@ class Meta:
                 ),
                 reverse_lazy("publications:create"),
             ),
+            "description": "Short description of this algorithm, max 1024 characters. This will appear in the info modal on the algorithm overview list.",
+            "detail_page_markdown": "<span class='text-danger'><i class='fa fa-exclamation-triangle'></i> This field will be deprecated. Please use the fields in the 'Algorithm description' tab to describe your algorithm instead.</span>",
+            "validation_and_performance": "If you have performance metrics about your algorithm, you can report them here. We recommend doing this in a table. <br>"
+            'Use a <a href = "https://www.tablesgenerator.com/markdown_tables"> markdown table generator</a>, or the following example to create your table:<br><br>'
+            "| | Metric 1 | Metric 2 |<br>"
+            "| --------- | --------- | -------- |<br>"
+            "| group 1 | 60% | 0.58 |<br>"
+            "| group 2 | 71% | 0.72 |<br>",
+            "mechanism": "Provide a short technical description of your algorithm. Think about the following aspects: <br>"
+            "- Target population: What clinical population does your algorithm target? <br>"
+            "- Algorithm description: Please provide a brief description of the methods of your algorithm.<br>"
+            "- Inputs and Outputs: The inputs and outputs your algorithm accepts and produces are automatically listed on the information page. <br>&nbsp;&nbsp;&nbsp;"
+            "Use this space here to provide additional details about them, if you wish.",
+            "common_error_messages": "Describe common error messages a user might encounter when trying out your algorithm and provide solutions for them. <br>"
+            "You might want to consider listing them in a table like this:<br><br>"
+            "| Error message | Solution | <br>"
+            "| --------- | ----------- | <br>"
+            "| error 1 | solution 1| <br>",
         }
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.helper = FormHelper(self)
+        self.helper.attrs = {"novalidate": "novalidate"}
```

Comment:
This is necessary to make validation work on the tabbed form. Required fields cannot be validated when a user submits the form from another form tab. This disables the standard validation. Instead we now validate the fields server-side (see clean method). I think the alternative here would have been to make the required fields optional, but that didn't seem like a better idea. A third option would be to split the form up into 2 seperate forms (then we'd no longer have the tabs) or put everything on one page (but that makes the form very long).

Reply:
I think the best thing to do here is to split this up into two forms, one for the algorithm with only the very basic data, and another for the algorithm metadata. Otherwise, you are having to duplicate a lot of work that Django should give you out of the box with form validation.